app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,45 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.encrypted[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.encrypted[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def authenticate
+    unless logged_in?
```

Comment:
I think it would be easier to read like:
```ruby
def authenticate
  redirect_to logint_path if current_user.nil? && !logged_in?
end
```

You can use an inline unless, but they are harder to follow when you have `unless` and `&&` or `||`